### ISSUE TITLE
SimpleInjector.ActivationException: 'No registration for type IRuntimePlatform could be found.'

### ISSUE THREAD

**Issue by hernanrz** [couchbase/couchbase-lite-net#1479]
When attempting to specify a directory for CBLite, an error is thrown 
Like so:

```cs
var dbconfig = new DatabaseConfiguration() {
  Directory =  Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "MyApp")
};
```

Just trying to instantiate DatabaseConfiguration triggers the rror

```
SimpleInjector.ActivationException: 'No registration for type IRuntimePlatform could be found.'
```

### Library Version
3.0.2

### .NET Runtime

.NET 6

### Operating System / Device Details

Windows 11 Insider Edition

### Expected behavior

Should be able to specify custom directory

**Comment by hernanrz** [comment]
I managed to work around the error by:

1. Creating a class that implements IRuntimePlatform

```cs
using Couchbase.Lite.DI;

internal class RuntimePlatform : IRuntimePlatform
    {
        public string OSDescription
        {
            get
            {
                return "Desktop";

            }
        }

        public string HardwareName
        {
            get
            {
                return "NET 6";
            }
        }
    }
```
    
    Then before calling CBLite in my code, I register the class:
    
 ```cs
 using Couchbase.Lite.DI;

            if (!RegisteredRuntimePlatform)
            {
                Service.Register<IRuntimePlatform>(new RuntimePlatform());
                RegisteredRuntimePlatform = true;
            }
```
    

It's very inelegant but it avoids the error

**Comment by jhimes144** [comment]
Also having the same issue, workaround provided fixes issue.

**Comment by borrrden** [comment]
It's interesting that this is the particular error that happens, and sorry that I didn't notice this ticket.  However .NET 6 is still, as of yet, unsupported.  3.1 will be the first release to support it.

**Comment by borrrden** [comment]
That being said I tried this with 3.0.8

```
using Couchbase.Lite;

var dbconfig = new DatabaseConfiguration()
{
    Directory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "MyApp")
};
using var db = new Database("foo", dbconfig);
Console.WriteLine(db.Path);
```

I didn't see any errors like this.  Can you confirm the reproduction steps?

**Comment by borrrden** [comment]
I tried just now with .net 4.7.2 and 3.0.8, and I do not see the issue.  Unrelated, but "pathHere" seems suspicious to me.  That should be the database name, not a path.

**Comment by borrrden** [comment]
Just for sanity sake, how are you observing this error?  I believe internally this exception is thrown by SimpleInjector but then caught and handled so it doesn't actually affect the program flow.  If you are just hitting it with a debugger set to halt on all exceptions it may trick you into thinking there is actually an issue where there is not.  `IRuntimePlatform` is not a required interface.  Couchbase Lite will try to instantiate it, but if it does not exist then it will fall back to some .NET runtime defaults.

**Comment by jhimes144** [comment]
@borrrden 

I think you might be onto something. I have another problem that is crashing our app when we move to 3x that is unrelated to this. I had my debugger set to recording non-user code exceptions and I think what I was seeing was the second problem silently killing the app after I continued through this exception, so I associated this exception as my app killer.

Since I turned off non-user code checks, I only get the unrelated error. Let me keep poking at this and I will get back to you.
